### PR TITLE
add support for regional disks

### PR DESCRIPTION
**Pull request by danawillow**
This PR adds support for regional disks, along with a few other changes to make them work correctly:
* adds `min_size` attribute that can be added to arrays. This is only used in Terraform, since `max_size` is only used in Terraform
* turns `sourceSnapshot` and `users` into ResourceRefs
* Adjusted TF flatteners so self links are always stored as v1

Even though RegionDisk is available in the v1 API, I opted to list it as beta since the feature itself is in beta.

-----------------------------------------------------------------
# [all]
Make disk snapshots and users into resourcerefs
## [terraform]
Add google_compute_region_disk resource.

### CODE REVIEW

**Comment by danawillow** [comment]
@rambleraptor for ansible changes and below question, @ndmckinley for the rest.

open q: Does it make sense to have users be a resourceref since it's an output field? For TF it does, since a ResourceRef there just means "handle this like a URL", but I'm not sure about the other providers.

**Comment by ndmckinley** [comment]
It does not make sense to do that in puppet to the best of my knowledge.  But if it made it through the unit tests, I suppose it doesn't really break anything.

File: .ci/magic-modules/generate-terraform.sh [hunk 1]
```diff
@@ -30,7 +30,7 @@ bundle exec compiler -p products/resourcemanager -e terraform -o "${GOPATH}/src/
 bundle exec compiler -p products/redis -e terraform -o "${GOPATH}/src/github.com/terraform-providers/terraform-provider-google/"
 
 # Resources that were already using beta APIs before they started being autogenerated
-bundle exec compiler -v beta -p products/compute -t Address,Subnetwork,ForwardingRule -e terraform -o "${GOPATH}/src/github.com/terraform-providers/terraform-provider-google/"
+bundle exec compiler -v beta -p products/compute -t Address,Subnetwork,ForwardingRule,RegionDisk -e terraform -o "${GOPATH}/src/github.com/terraform-providers/terraform-provider-google/"
```
Comment (ndmckinley) [inline .ci/magic-modules/generate-terraform.sh]: This is getting a bit untenable - any ideas?

File: products/compute/helpers/terraform/disk_overrides.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+# limitations under the License.
+id: !ruby/object:Provider::Terraform::PropertyOverride
+  exclude: true
+sizeGb: !ruby/object:Provider::Terraform::PropertyOverride
+  name: size
+  default_from_api: true
+sourceSnapshot: !ruby/object:Provider::Terraform::PropertyOverride
+  name: snapshot
+  description: |
+    {{description}}* snapshot
```
Comment (ndmckinley) [inline products/compute/helpers/terraform/disk_overrides.yaml]: What does `{{description}} * snapshot` do?

File: .ci/magic-modules/generate-terraform.sh (hunk 1 above)
Comment (danawillow) [reply .ci/magic-modules/generate-terraform.sh]: In the future, we won't have this list at all- it'll put all resources at their beta versions into terraform-provider-google-beta, so I'm pretty ok with leaving it as-is as a temporary thing. I can pretty easily factor out the list into a variable though if you'd like.

File: products/compute/helpers/terraform/disk_overrides.yaml (hunk 2 above)
Comment (danawillow) [reply products/compute/helpers/terraform/disk_overrides.yaml]: https://github.com/terraform-providers/terraform-provider-google/pull/1755/files#diff-a1e4f4837e83c9ac24d81e8adec94a89R149

(terraform is the only one that also accepts just the resource name)

File: .ci/magic-modules/generate-terraform.sh (hunk 1 above)
Comment (ndmckinley) [reply .ci/magic-modules/generate-terraform.sh]: I could imagine some solutions to keep you from having to type it at all (we never want to not generate something at beta if it's got beta support in `api.yaml`), but if it's not bugging you let's live with it for now.

File: products/compute/helpers/terraform/disk_overrides.yaml (hunk 2 above)
Comment (ndmckinley) [reply products/compute/helpers/terraform/disk_overrides.yaml]: :man_facepalming: of course.

**Pull request merged**